ipa-core/src/cli/playbook/hybrid.rs

```diff
@@ -0,0 +1,135 @@
+#![cfg(all(feature = "web-app", feature = "cli"))]
+use std::{
+    cmp::min,
+    time::{Duration, Instant},
+};
+
+use futures_util::future::try_join_all;
+use serde::{Deserialize, Serialize};
+use tokio::time::sleep;
+
+use crate::{
+    ff::{Serializable, U128Conversions},
+    helpers::{
+        query::{HybridQueryParams, QueryInput, QuerySize},
+        BodyStream,
+    },
+    net::{Helper, IpaHttpClient},
+    protocol::QueryId,
+    query::QueryStatus,
+    secret_sharing::{replicated::semi_honest::AdditiveShare, SharedValue},
+    test_fixture::Reconstruct,
+};
+
+/// # Panics
+/// if results are invalid
+#[allow(clippy::disallowed_methods)] // allow try_join_all
+pub async fn run_hybrid_query_and_validate<HV>(
+    inputs: [BodyStream; 3],
+    query_size: usize,
+    clients: Vec<[IpaHttpClient<Helper>; 3]>,
+    query_id: QueryId,
+    query_config: HybridQueryParams,
+) -> HybridQueryResult
+where
+    HV: SharedValue + U128Conversions,
+    AdditiveShare<HV>: Serializable,
+{
+    let mpc_time = Instant::now();
+
+    // for now, submit everything to the leader. TODO: round robin submission
+    let leader_clients = &clients[0];
+    try_join_all(
+        inputs
+            .into_iter()
+            .zip(leader_clients)
+            .map(|(input_stream, client)| {
+                client.query_input(QueryInput {
+                    query_id,
+                    input_stream,
+                })
+            }),
+    )
+    .await
+    .unwrap();
```

Comment:
@akoshelev I ended up needing to create a new function here that could take `Vec<[IpaHttpClient<Helper>; 3]>`. right now it submits everything to the leader, but this seems like the natural place to do the round robin submission.